pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/DistinctFromTransformFunction.java

```diff
@@ -0,0 +1,140 @@
+  static private BitSet getNullBits(@Nullable PeekableIntIterator nullValueVectorIterator, int length, int[] docIds) {
+    BitSet nullBits = new BitSet(length);
+    if (nullValueVectorIterator == null) {
+      return nullBits;
+    }
+    int currentDocIdIndex = 0;
+    while (nullValueVectorIterator.hasNext() & currentDocIdIndex < length) {
+      nullValueVectorIterator.advanceIfNeeded(docIds[currentDocIdIndex]);
+      currentDocIdIndex = Arrays.binarySearch(docIds, currentDocIdIndex, length, nullValueVectorIterator.next());
+      if (currentDocIdIndex >= 0) {
+        nullBits.set(currentDocIdIndex);
+        currentDocIdIndex++;
+      } else {
+        currentDocIdIndex = -currentDocIdIndex - 1;
+      }
+    }
+    return nullBits;
+  }
+
+  // Returns null iterator based on nullValueVectorIterator.
+  // If null value is not enabled, which means nullValueVectorIterator can be null, this function will return a null.
+  @Nullable
+  static private PeekableIntIterator getNullValueIterator(Map<String, DataSource> dataSourceMap,
+      TransformFunction transformFunction, PeekableIntIterator nullValueVectorIterator) {
+    String columnName = ((IdentifierTransformFunction) transformFunction).getColumnName();
+    NullValueVectorReader nullValueVectorReader = dataSourceMap.get(columnName).getNullValueVector();
+    if (nullValueVectorIterator != null) {
+      return nullValueVectorReader.getNullBitmap().getIntIterator();
+    }
+    return null;
+  }
+
+  @Override
+  public String getName() {
+    if (_getDistinct == 1) {
+      return TransformFunctionType.IS_DISTINCT_FROM.getName();
+    }
+    return TransformFunctionType.IS_NOT_DISTINCT_FROM.getName();
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+    super.init(arguments, dataSourceMap);
+    if (!(_leftTransformFunction instanceof IdentifierTransformFunction)
+        || !(_rightTransformFunction instanceof IdentifierTransformFunction)) {
+      throw new IllegalArgumentException("Only column names are supported in DistinctFrom transformation.");
+    }
+    _leftNullValueVectorIterator =
+        getNullValueIterator(dataSourceMap, _leftTransformFunction, _leftNullValueVectorIterator);
+    _rightNullValueVectorIterator =
+        getNullValueIterator(dataSourceMap, _rightTransformFunction, _rightNullValueVectorIterator);
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return BOOLEAN_SV_NO_DICTIONARY_METADATA;
+  }
+
+  @Override
+  public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
+    int length = projectionBlock.getNumDocs();
+    if (_results == null || _results.length < length) {
+      _results = new int[length];
+    }
+    _results = super.transformToIntValuesSV(projectionBlock);
+    int[] docIds = projectionBlock.getDocIds();
+    BitSet leftNull = getNullBits(_leftNullValueVectorIterator, length, docIds);
```

Comment:
We can get the null bitmap using `BlockValSet.getNullBitmap()`. No need to construct it from the data source.

Reply:
Done..